Realm/RLMArray.mm

```diff
@@ -437,3 +438,23 @@ - (instancetype)reversedSortDescriptor {
 }
 
 @end
+
+@interface RLMArray (Handover) <RLMThreadConfined_Private>
+@end
+
+@implementation RLMArray (Handover)
+
+- (realm::AnyThreadConfined)rlm_handoverData {
+    @throw RLMException(@"Unexpected handover of unmanaged `RLMArray`");
```

Comment:
`REALM_UNREACHABLE();` calls terminate with an appropriate message and should be marked as noreturn.